### PR TITLE
Link to homeUrl from site editor view menu.

### LINKED ISSUE #45474: Site Editor: View > View Site opens incorrect URL.
### Description

In the Site Editor, the `View > View site` drop down opens the URL of the WordPress install (`site_url()`) rather than the URL of the website (`home_url()`).

Follow up to https://github.com/WordPress/gutenberg/pull/42331

### Step-by-step reproduction instructions

1. Create a [sub-directory install](https://wordpress.org/support/article/giving-wordpress-its-own-directory/) of WordPress 
2. Visit the site editor
3. From the view dropdown, click view site
4. Observe the window opens with the WordPress install sub directory included.

Note: I discovered this using [chassis/chassis](https://github.com/chassis/chassis), if you have both Vagrant and VirtualBox installed, you may wish to use Chassis to set up an environment quickly.

### Screenshots, screen recording, code snippet

Please ignore the apparently incorrect WP version in the site's title. I noticed this while testing during the WordPress 6.1 release.

view-site

### Environment info

- PHP 7.4
- WordPress 6.1.0
- Gutenberg plugin is not installed
- Home URL: wp59.local
- WP URL: wp59.local/wp

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by peterwilsoncc**
## What?

Replaces the siteURL link in the full site editor view menu with the homeUrl.

## Why?

siteUrl is a reference to the WordPress install, ie the folder containing wp-cron.php, wp-mail.php, etc. homeUrl is the home page of the website. On sites with WordPress installed in a sub-directory, the two values can differ.

## How?

I've added `@wordpress/core-data` as a dependency of the site editor header and used that to locate the homeUrl. I suspect it could be added equally well the the existing settings but I was unsure how to go about that.

Fixes #45474.

## Testing Instructions
1. Create a [sub-directory install](https://wordpress.org/support/article/giving-wordpress-its-own-directory/) of WordPress 
2. Visit the site editor
3. From the view dropdown, click view site

This should open the site's home page rather than the sub-directory containing the install/

### CODE REVIEW

File: packages/edit-site/src/components/header-edit-mode/index.js [hunk 1]
```diff
@@ -50,22 +51,25 @@ export default function Header( { showIconLabels } ) {
...
 		} = select( editSiteStore );
 		const { getShortcutRepresentation } = select( keyboardShortcutsStore );
 		const { __unstableGetEditorMode } = select( blockEditorStore );
 
 		const postType = getEditedPostType();
 
+		const {
+			getUnstableBase, // Site index.
+		} = select( coreStore );
+
```
Comment (peterwilsoncc) [inline packages/edit-site/src/components/header-edit-mode/index.js:72]: This is a 100% copy-paste from `packages/block-library/src/home-link/edit.js` -- let me know if there is a better way. I am sure there is.

File: packages/edit-site/src/components/header-edit-mode/index.js [hunk 2]
```diff
@@ -75,8 +79,8 @@ export default function Header( { showIconLabels } ) {
 				'core/edit-site/toggle-list-view'
 			),
 			isVisualMode: getEditorMode() === 'visual',
-			settings: getSettings(),
 			blockEditorMode: __unstableGetEditorMode(),
+			homeUrl: getUnstableBase()?.home,
```
Comment (Mamaduka) [inline packages/edit-site/src/components/header-edit-mode/index.js:83]: This is much better. We were only using the URL setting, so there was no need to return the whole object.

**Review by Mamaduka — APPROVED** [review]
Thank you, @peterwilsoncc!

The fix works as expected 👍

**Comment by peterwilsoncc** [comment]
Replaced by #45476, since merged.

@Mamaduka Thanks for following this up, your fix is much nicer.

**Comment by Mamaduka** [comment]
@peterwilsoncc, the https://github.com/WordPress/gutenberg/pull/45476 solved the issue for the Site Logo block. We still need to resolve this for the "View" URL.

**Pull request merged**